st2common/st2common/util/output_schema.py

```diff
@@ -51,6 +52,23 @@ def _validate_action(action_schema, result, output_key):
     schema.validate(final_result, action_schema, cls=schema.get_validator("custom"))
 
 
+def mask_secret_output(value, output_schema_result):
+    value = copy.deepcopy(value)
+    for key in value["action"]["output_schema"]:
+        if (
+            value.get("action", {})
+            .get("output_schema", {})
+            .get(key)
+            .get("secret", False)
+        ):
+            if output_schema_result.get("result"):
+                output_schema_result["result"][key] = MASKED_ATTRIBUTE_VALUE
+            if output_schema_result.get("body"):
+                output_schema_result["body"][key] = MASKED_ATTRIBUTE_VALUE
```

Comment:
@mahesh-orch Instead of hard coding "result" and "body" and using if else, how come you are not using the output key from the runner model dynamically like https://github.com/StackStorm/st2/blob/master/st2common/st2common/util/action_db.py#L247?  I mentioned this in my last comment. What you've done here only works for python and http runner. Workflow action will fail. Please add a unit test for the Orquesta runner and you will see.